### PR TITLE
add mempolicy for split mode

### PR DESCRIPTION
**Pull request by lplewa**

### CODE REVIEW

File: include/umf/mempolicy.h [hunk 1]
```diff
@@ -25,17 +25,30 @@ typedef enum umf_mempolicy_membind_t {
     /// Bind memory to namespace
     UMF_MEMPOLICY_BIND,
     /// Prefer memory from namespace but fallback to other memory if not available
-    UMF_MEMPOLICY_PREFERRED
+    UMF_MEMPOLICY_PREFERRED,
+    /// Allocation will be split evenly across nodes specified in nodemask.
+    /// umf_numa_split_partition_t can be passed in umf_os_memory_provider_params_t structure
```
Comment (lukaszstolarczuk) [inline include/umf/mempolicy.h:30]: did you mean `umf_mempolicy_split_partition_t`? and used in `umfMempolicySetCustomSplitPartitions`?

File: test/memspaces/mempolicy.cpp [hunk 2]
```diff
@@ -109,5 +112,63 @@ TEST_F(test, mempolicyInterleavePartSize) {
...
+    umfMemoryProviderDestroy(hProvider);
+}
+
+TEST_F(test, mempolicyCustomSplit) {
+    umf_memory_provider_handle_t hProvider = nullptr;
+    umf_mempolicy_handle_t hPolicy = nullptr;
+
+    umf_result_t ret = umfMempolicyCreate(UMF_MEMPOLICY_SPLIT, &hPolicy);
+    ASSERT_EQ(ret, UMF_RESULT_SUCCESS);
+    umf_mempolicy_split_partition_t part[] = {{1, 0}, {1, 0}};
```
Comment (lukaszstolarczuk) [inline test/memspaces/mempolicy.cpp:150]: shouldn't we test more various patterns (similarly to your prev. PR)? at best with randomization...?

Comment (lplewa) [reply test/memspaces/mempolicy.cpp:150]: my idea is that os_provider is allready tested - so here i only test copy one struct do different one - i do not want to copy all numa tests just do change how we create osprovider. This is why tests in this file are quite simple

**Review by bratpiorka — APPROVED** [review]

File: include/umf/mempolicy.h [hunk 3]
```diff
@@ -25,9 +25,22 @@ typedef enum umf_mempolicy_membind_t {
     /// Bind memory to namespace
     UMF_MEMPOLICY_BIND,
     /// Prefer memory from namespace but fallback to other memory if not available
-    UMF_MEMPOLICY_PREFERRED
+    UMF_MEMPOLICY_PREFERRED,
+    /// Allocation will be split evenly across nodes specified in nodemask.
+    /// umf_mempolicy_split_partition_t can be used to specify different distribution
```
Comment (lukaszstolarczuk) [inline include/umf/mempolicy.h:30]: `different distribution to specify other distribution`...? re-phrase I guess...?

File: src/mempolicy.c [hunk 4]
```diff
@@ -45,6 +48,22 @@ umf_result_t umfMempolicySetInterleavePartSize(umf_mempolicy_handle_t policy,
...
+    policy->ops.interleave.part_size = partSize;
+    return UMF_RESULT_SUCCESS;
+}
+
+umf_result_t
+umfMempolicySetCustomSplitPartitions(umf_mempolicy_handle_t policy,
+                                     umf_mempolicy_split_partition_t *partList,
+                                     size_t partListLen) {
+    if (policy == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
```
Comment (lukaszstolarczuk) [inline src/mempolicy.c:60]: we could have tests for these 2 negative scenarios

Comment (lplewa) [reply src/mempolicy.c:60]: Done

File: include/umf/mempolicy.h (hunk 3 above)
Comment (lplewa) [reply include/umf/mempolicy.h]: Done

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**